core/storage/suggestion/gae_models.py

```diff
@@ -2394,28 +2398,23 @@ def fetch_page(
 
         sorted_results: List[
             TranslationSubmitterTotalContributionStatsModel] = []
-        today = datetime.date.today()
-
-        if max_days_since_last_activity is not None:
-            last_date = today - datetime.timedelta(
-                days=max_days_since_last_activity)
-            next_offset = offset
-            while len(sorted_results) < page_size:
-                result_models: Sequence[
-                    TranslationSubmitterTotalContributionStatsModel] = (
-                    sort_query.fetch(
-                        NUM_MODELS_PER_FETCH, offset=next_offset))
-                if not result_models:
+
+        next_offset = offset
+        while len(sorted_results) < page_size:
+            result_models: Sequence[
+                TranslationSubmitterTotalContributionStatsModel] = (
+                sort_query.fetch(
+                    NUM_MODELS_PER_FETCH, offset=next_offset))
+            if not result_models:
+                break
+            for result_model in result_models:
+                if len(sorted_results) == page_size:
                     break
-                for result_model in result_models:
-                    next_offset += 1
-                    if result_model.last_contribution_date >= last_date:
-                        sorted_results.append(result_model)
-                        if len(sorted_results) == page_size:
-                            break
-        else:
-            sorted_results = list(sort_query.fetch(page_size, offset=offset))
-            next_offset = offset + len(sorted_results)
+                next_offset += 1
+                if (((result_model.first_contribution_date) <= (
+                    start_date)) and (
+                    result_model.last_contribution_date >= end_date)):
```

Comment:
This looks a bit strange. 
1. Too many parenthesis
2. The condition seems to be incorrect. What if the last contribution is between start and end date? If I'm right here, please add tests for this situation

Reply:
Yes, you are right.
Fixed the logic.